File: spring-web-java/src/main/java/spring/web/java/presentation/dto/response/PermissionResponse.java
package spring.web.java.presentation.dto.response;

import spring.web.java.domain.model.entity.Permission;

public record PermissionResponse(
	Long id,
	String name
) {
	public PermissionResponse(Permission permission) {
		this(permission.getId(), permission.getName());
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/response/PermissionResponse.kt
package spring.web.kotlin.presentation.dto.response

import spring.web.kotlin.domain.model.entity.Permission

data class PermissionResponse(
    val id: Long,
    val name: String
) {
    constructor(permission: Permission) : this(
        id = checkNotNull(permission.id),
        name = permission.name
    )
}